docs/development/index.rst

```diff
@@ -1,20 +1,238 @@
+
+
+DevOps
+------
+
+Before you go and improve our tooling, please start by first opening a new issue
+and discussing your idea with us. We ask this, because we want to make sure that
+there is at least one maintainer familiar with the technology you want to use.
+If anything breaks in the CI/CD pipeline, it will have a direct impact on our
+release cycle, and we would like to make sure that at least one maintainer is
+able to step in and fix it in a timely fashion.
+
+**Installation**
 
-Developer Installation
-----------------------
+While you may be able to complete some devops related contributions in GitHub itself,
+a local setup will perform better in most situations::
 
-For developers, we provide a simple mechanism to allow importing
-imageio from the cloned repository <https://github.com/imageio/imageio.git>::
+    # 1. Fork the ImageIO repository
 
-    git clone https://github.com/imageio/imageio.git
+    git clone https://github.com/<replace_by_your_username>/imageio.git
     cd imageio
-    pip install -e .[dev]
+    pip install -e .
+
+    # create and checkout a new branch for your contribution
+    # Note: replace <branch_name> with a descriptive, but short, name
+    git checkout -b <branch_name>
+    git push --set-upstream origin <branch_name>
+
+**Developing the PR**
+
+Do your changes, and remember to commit frequently. If you want to test your
+changes against our CI/CD provider (GH Actions), you can submit a draft PR,
+which we will review and run. Alternatively, you configure your local fork to
+execute actions and develop against it. You can find our worflows in the
+``.github`` folder.
+
+*Continous Integration (CI)*: The core of our CI is a matrix test across all
+major OSes and supported python versions running a testsuite based on
+``pytest``. Additionally, we track linting errors (black + flake8), coverage
+(codecov), and have readthedocs build a preview of the documentation.
+
+*Continous Deployment (CD)*: Our continous deployment pipeline checks weekly
+(Monday night, EU time) for any changes to the repository. If there are any, it
+uses python-semantic-release to figure out if the version should bump and a new
+release should be made. If so, all tests are run against the master branch, the
+version is bumped, and a new release is made and published to GitHub and PyPI.
+Further, we have a feedstock on conda-forge which manages releases to conda.
+
+**Submitting the PR**
+
+Once you are happy with your changes, push your changes to your local fork, head
+over to the `main repo <https://github.com/imageio/imageio>`_ and create a new
+PR. In the description, briefly summarize what the changes are about (there is
+no fixed format) and add a reference to the issue in which we discussed them.
+From there, we will help you by reviewing your changes, discussing them and
+eventually merging your work into ImageIO.
+
+Thank you for contributing!
+
+
+
+Code Contributions
+------------------
```

Comment:
Maybe move this up? I at least expect code contributions to be more "popular" than devops.